internal/tasks/fleet_validate.go

```diff
@@ -56,16 +52,23 @@ type FleetValidateLogic struct {
 	log         logrus.FieldLogger
 	store       store.Store
 	resourceRef ResourceReference
+	repoNames   []string
 }
 
 func NewFleetValidateLogic(taskManager TaskManager, log logrus.FieldLogger, store store.Store, resourceRef ResourceReference) FleetValidateLogic {
 	return FleetValidateLogic{taskManager: taskManager, log: log, store: store, resourceRef: resourceRef}
 }
 
 func (t *FleetValidateLogic) CreateNewTemplateVersionIfFleetValid(ctx context.Context, fleet *api.Fleet) error {
-	err := t.validateFleetTemplate(ctx, fleet)
-	if err != nil {
-		return fmt.Errorf("validating fleet: %w", err)
+	validationErr := t.validateFleetTemplate(ctx, fleet)
```

Comment:
If there is a validation error why do we continue execution and mutate the store or should we return?